main/pom.xml

```diff
@@ -348,12 +348,19 @@
       <version>${jetty.version}</version>
     </dependency>
 
-    <!-- Language detector api -->
+    <!-- Language detector -->
     <dependency>
       <groupId>com.optimaize.languagedetector</groupId>
       <artifactId>language-detector</artifactId>
       <version>0.6</version>
     </dependency>
+
+    <!-- date parser -->
+    <dependency>
+      <groupId>com.github.sisyphsu</groupId>
+      <artifactId>dateparser</artifactId>
+      <version>1.0.10</version>
+    </dependency>
```

Comment:
I am actually a bit worried about including this dependency. Although it is actively maintained, it does not seem to be used a lot, so it is likely that we get surprises with it: https://mvnrepository.com/artifact/com.github.sisyphsu/dateparser
Also, the license it declares on GitHub (MIT) is not the one included in the Maven Metadata (Apache license).

Reply:
Additionally, it was implemented as a showcase for the underlying RE implementation, rather than as something driven by user requirements and it also appears to be English only for month & day names.

Another possibility might be a JNA wrapper to https://github.com/araddon/dateparse which seems to be a little bit more general and much more popular.

However, one of the things that I think to current date guesser has shown us is that something which works *most* of the time is kind of a recipe for unforeseen problems and complaints.